### DISCUSSION TITLE
Using x-for with x-model and persist plugin problem

### DISCUSSION THREAD

**Discussion by dgvirtual** [alpinejs/alpine#3196] [1. Help] [answered]
I have this select in my app: 

```
<select id="kasKreipesi" class="form-select" x-model="data.kasKreipesi">
    <option></option>
    <template x-for="initiator in who">
        <option x-text="initiator"></option>
    </template>
</select>
```

And I am using persist plugin to store the data in the local storage. So in my component, within which the select element is, I have the data that is used to generate the select options and the property data.kasKreipesi : 

```
<div x-data="{
    who: [
        'One',
        'Two',
        'Three',
    ],
   data: {
        kasKreipesi: $persist(''),
   }
}">

```

While I had the options of select hardcoded, it all worked fine: the page would remember the selected item after reload. But when I started using x-for to generate the options on the fly from the list in x-data, it stopped working, the list is generated fine, but the selected option is not reselected. Thought I can still see the correct data in LocalStorage. 

What could be the cause and how do I solve it? I am using the latest Alpine JS with the latest (persist@3.x.x) persist plugin.

**Comment by ekwoka** [comment]
Well, for starters, you should put a `:key` on the `x-for`

I believe this is an issue where the model is processed before the `x-for` so the value doesn't exist when the model starts.

So, you can probably use `x-init` on the options to ensure the correct one is selected when it loads.

**Reply by dgvirtual** [reply]
Thank you for the response, I appreciate your help.

> Well, for starters, you should put a `:key` on the `x-for`

I realize that is the standard, but did not bother in this case, since this particular select was not to be modified during runtime. 

> I believe this is an issue where the model is processed before the `x-for` so the value doesn't exist when the model starts.
> So, you can probably use `x-init` on the options to ensure the correct one is selected when it loads.

I would like to try that for the sake of learning, but I don't quite understand what you mean or how to do that. Could you elaborate please?

**Comment by SimoTod** [comment, marked as answer]
This👆
Or you can add `:selected="initiator == data.kasKreipesiVariantai"` to your generated options

**Reply by dgvirtual** [reply]
This worked, thank you! Where is that :selected documented? Don't see...

**Reply by SimoTod** [reply]
It's the short form for x-bind:selected
Selected is just the standard html attribute
